hle-survey/transformation/src/main/scala/org/broadinstitute/monster/dap/HealthTransformations.scala

```diff
@@ -42,173 +38,248 @@ object HealthTransformations {
-            conditionCauseOtherDescription =
-              if (isCauseKnown && (conditionCause.getOrElse(None) == 98))
-                rawRecord.getOptional("hs_dx_eye_cause_other")
-              else None
-          )
-        }
-        case (disease, categorical) =>
-          createHealthConditionRow(rawRecord, s"dx_${disease}", eyeDiseaseCondition, categorical)
-      }
-    } else None
+  def processHealthCondition(
+    rawRecord: RawRecord,
+    healthCondition: HealthCondition
+  ): Iterable[HlesHealthCondition] = {
+    val congenital: Option[HlesHealthCondition] = healthCondition.congenital.flatMap {
+      case (target, dependent) =>
+        // if congenital, then check hs_congenital_yn && healthCondition.congenital._2
+        if (rawRecord.getBoolean("hs_congenital_yn") &&
+            rawRecord.getBoolean(s"hs_${dependent}_yn"))
+          // congenital other
+          if (healthCondition.isOther)
+            createHealthConditionRow(
+              rawRecord,
+              healthCondition.congenital.get._1,
+              conditionTypes.apply(healthCondition.conditionType),
+              conditions.apply(healthCondition.condition),
+              isCongenital = true,
+              conditionOtherDescription = rawRecord.getOptional(s"hs_${target}_spec")
+            )
+          // congenital general
+          else if (rawRecord.getBoolean(s"hs_${target}"))
+            createHealthConditionRow(
+              rawRecord,
+              target,
+              conditionTypes.apply(healthCondition.conditionType),
+              conditions.apply(healthCondition.condition),
+              isCongenital = true
+            )
+          else None
+        else None
+    }
+    val nonCongenital: Option[HlesHealthCondition] = healthCondition.nonCongenital.flatMap {
+      case (target, dependent) =>
+        if (rawRecord.getBoolean(s"hs_${dependent}_yn"))
+          // non-congenital other
+          if (healthCondition.isOther)
+            createHealthConditionRow(
+              rawRecord,
+              target,
+              conditionTypes.apply(healthCondition.conditionType),
+              conditions.apply(healthCondition.condition),
+              conditionOtherDescription = rawRecord.getOptional(s"hs_${target}_spec")
+            )
+          // specific non-congenital blindness case
+          else if (healthCondition.condition == "blind") {
+            val isCauseKnown = rawRecord.getBoolean("hs_dx_eye_cause_yn")
+            val conditionCause =
+              if (isCauseKnown) rawRecord.getOptionalNumber("hs_dx_eye_cause") else None
+            createHealthConditionRow(
+              rawRecord,
+              target,
+              conditionTypes.apply(healthCondition.conditionType),
+              conditions.apply(healthCondition.condition),
+              conditionCause = conditionCause,
+              conditionCauseOtherDescription =
+                if (isCauseKnown && (conditionCause.getOrElse(false) == 98))
```

Comment:
`getOrElse(false)` and `98` never make sense to compare, one side or the other needs updated?

Reply:
ahaha yup ty